tests/publiccloud/sles4sap.pm

```diff
@@ -106,35 +118,53 @@ sub fence_node {
     $args{hostname} //= $self->{my_instance}->{instance_id};
 
     # Do the fencing!
-    record_info('Fencing info', 'Fencing done with systemctl poweroff');
-    $self->run_cmd(cmd => 'systemctl poweroff --force', timeout => 0);
-
-    # skip the one minute waiting
-    sleep 60;
-    my $start_time = time();
+    if (check_var('USE_FENCING', 'poweroff')) {
+        record_info('Fencing info', 'Fencing done with systemctl poweroff');
+        $self->run_cmd(cmd => 'systemctl poweroff --force', timeout => 0, quiet => 1);
+    } else {    # Default fencing uses CRM
+        record_info('Fencing info', 'Fencing done with crm');
+        $self->run_cmd(cmd => "crm -F node fence $args{hostname}", quiet => 1);
+    }
 
     # Wait till ssh disappear
-    while ((time() - $start_time) < $timeout) {
-        last unless (defined($self->{my_instance}->wait_for_ssh(timeout => 1, proceed_on_failure => 1)));
+    my $start_time = time;
+    while ((time - $start_time) < $timeout) {
+        last unless (defined($self->{my_instance}->wait_for_ssh(timeout => 1, proceed_on_failure => 1, quiet => 1)));
     }
-    my $fencing_time = time() - $start_time;
+    my $fencing_time = time - $start_time;
     die("Waiting for fencing failed!") unless ($fencing_time < $timeout);
     record_info('Fencing finished!', "Fencing done in ${fencing_time}s");
 
+    # We need to be sure that the server is stopped (if applicable)
+    if (check_var('PUBLIC_CLOUD_PROVIDER', 'EC2')) {
```

Comment:
you can use this function here https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/utils.pm#L52 ( and in other similar places )

Reply:
Thanks, I will check and use it.

Reply:
Done, and I also added the same function for Azure and GCE.